src/StepAggregator.php

```diff
@@ -192,4 +159,35 @@ public function getName()
     {
         return $this->name;
     }
+
+    /**
+     * Builds the pipeline
+     *
+     * @return callable
+     */
+    private function buildPipeline()
+    {
+        $nextCallable = function ($item) {
+            // the final callable is a no-op
+        };
```

Comment:
Interesting way to keep `$next` required and remove the need for `null` checks in all steps.